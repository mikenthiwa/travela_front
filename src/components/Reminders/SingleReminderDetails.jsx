import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { isEmpty } from 'lodash';
import Modal from '../modal/Modal';
import ReminderItem from './ReminderItem';
import SubmitArea from '../Forms/NewEmailTemplateForm/FormFieldSets/SubmitArea';
import Preloader from '../Preloader/Preloader';
import Utils from '../../helper/Utils';
import './SingleReminderDetails.scss';

class SingleReminderDetails extends Component {

  handleOnSend = (event) => {
    event.preventDefault();
    const { history, singleReminder: { data: { id } }  } = this.props;
    history.push(`/settings/reminders/edit/${id}`);
  }
  
  handleCancel (event, closeModal){
    event.preventDefault();
    closeModal();
  }

  renderReminderDetails(reminders, conditionName) {
    return (
      <div className="reminder_details">
        {reminders.map(
          (reminder, index) => (
            <ReminderItem 
              key={reminder.id} 
              reminder={reminder} 
              index={index + 1}
              conditionName={conditionName}
            />
          )
        )}
      </div>
    );
  }

  render () {
    const {
      closeModal, 
      modalType, 
      shouldOpen, 
      singleReminder: { data }, 
      activeDocument,
      conditionId
    } = this.props;
    const { conditionName, reminders } = data;
    return(
      <div className="reminder__deatails">
        <Modal
          customModalStyles="template-details"
          closeModal={closeModal} width="504px"
          visibility={(shouldOpen && (modalType === `reminder details${conditionId}`))
            ? 'visible' : 'invisible'}
          title={`${Utils.makeTitleCase(activeDocument)} Expiry Reminder`}>
          { isEmpty(data.reminders) ? <Preloader /> : (
            <Fragment>
              {this.renderReminderDetails(reminders, conditionName)}
              <hr />
              <div className="submit-area">
                <SubmitArea 
                  onCancel={(event) => this.handleCancel(event, closeModal)}
                  onSend={(event) => this.handleOnSend(event)}
                  send="Edit Reminder Condition"
                  reversed
                />
              </div>
            </Fragment>
          )}
        </Modal>
      </div>
    );
  }
}

SingleReminderDetails.propTypes = {
  closeModal: PropTypes.func,
  modalType: PropTypes.string,
  shouldOpen: PropTypes.bool,
  conditionId: PropTypes.number,
  singleReminder: PropTypes.object,
  activeDocument: PropTypes.string.isRequired,
  history: PropTypes.object.isRequired
};

SingleReminderDetails.defaultProps = {
  modalType: '',
  conditionId: null,
  closeModal: () => { },
  shouldOpen: false,
  singleReminder: {
    data: {
      reminders: []
    },
  },
};

export default SingleReminderDetails;
