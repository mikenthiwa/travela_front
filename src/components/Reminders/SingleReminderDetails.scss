.reminder__deatails {
  .template-details .modal-content {
    margin-bottom: 20px;
    hr {
      margin-left: -7.2%;
      box-sizing: border-box;
      height: 1px;
      width: 100%;
      border: 1px solid #F8F8F8;
      margin-bottom: 40px !important;
    }
  }
  .modal-title-bar .modal-title-text {
    height: 27px;
    width: 213px;
    color: #3359DB;
    font-size: 21px;

    line-height: 27px;
    @media only screen and (max-width: 340px) {
      .modal-title-word {
        font-size: 17px;
      }
    }
  }
  .reminder_details {
    margin: 3px auto;
    display: flex;
    flex-direction: column;
    text-align: left;
    overflow-y: scroll;
    margin-top: 30px;
    p {
      height: 20px;
      width: 78px;
      color: #4F4F4F;
      font-family: "DIN Pro";
      font-size: 15px;
      line-height: 20px;
      @media screen and (max-width: 479px) {
        font-size: 14px;
      }
    }
    .reminder__condition {
      height: 68px;
      width: 100%;
      display: flex;
      justify-content: center;
      border-radius: 2px;
      background-color: #F6F6F6;
      flex-direction: column;
      margin-bottom: 42px;
      p, span {
        padding-left: 8px;
      }
      &:last-of-type {
        margin-bottom: 20px;
      }
      .frequency {
        height: 18px;
        width: 182px;
        color: #878787;
        font-size: 14px;
        line-height: 18px;
      }
    }
  }
  .submit-area {
    .bg-btn--active {
      margin-right: 0;
    }
  }
}

@media only screen and (max-width: 571px) {
  .delete-checklist-item {
    .modal-title-bar {
      .modal-title-word {
        padding-top: 15px;
      }
    }
  }
}
