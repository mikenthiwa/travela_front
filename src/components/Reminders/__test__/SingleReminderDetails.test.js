import React from 'react';
import SingleReminderDetails from '../SingleReminderDetails';

const props = {
  conditionId: 1,
  activeDocument: 'passport',
  singleReminder: {
    data: {
      conditionName: 'Passport Expiry',
      id: 1,
      reminders: [
        {
          frequency: '2 Months',
          id: 1,
          reminderTemplateId: 1,
        },
        {
          frequency: '3 Weeks',
          id: 2,
          reminderTemplateId: 1,
        }
      ],
    },
  },
  closeModal: jest.fn(),
  handleCancel: jest.fn(),
  handleOnSend: jest.fn(),
  shouldOpen: true,
  modalType: 'reminder details1',
  history: {
    push: jest.fn()
  }
};

describe('<SingleReminderDetails />', ()=> {
  it('should render correctly', () =>{
    const wrapper = mount(<SingleReminderDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should display reminder with two frequncy', () =>{
    const wrapper = mount(<SingleReminderDetails {...props} />);
    const reminders = wrapper.find('ReminderItem');
    expect(reminders).toHaveLength(2);
    expect(wrapper.find('Modal .frequncy_count').first().text()).toBe('Reminder 1');
  });

  it('should not open reminder details modal when \'shouldOpen\' is false', () =>{
    const wrapper = mount(<SingleReminderDetails {...props} shouldOpen={false} />);
    const reminders = wrapper.find('ReminderItem');
    expect(reminders).toHaveLength(0);
  });

  it('should open reminder details modal with reminder details \'shouldOpen\' is true', () =>{
    const wrapper = mount(<SingleReminderDetails {...props} shouldOpen />);
    const reminderDeailsTitle = wrapper.find('ReminderItem .frequncy_count');
    const reminderDetailsFrequency = wrapper.find('ReminderItem .frequency');
    expect(reminderDeailsTitle.at(0).text()).toBe('Reminder 1');
    expect(reminderDeailsTitle.at(1).text()).toBe('Reminder 2');
    expect(reminderDetailsFrequency.at(0).text())
      .toBe('To be sent 2 Months to Expiry');
    expect(reminderDetailsFrequency.at(1).text())
      .toBe('To be sent 3 Weeks to Expiry');
  });

  it('should close reminder details modal when \'closeModal\' function is called', () =>{
    const wrapper = mount(<SingleReminderDetails {...props} shouldOpen />);
    const handleCancel = jest.spyOn(wrapper.instance(), 'handleCancel');
    const closeModal = jest.spyOn(wrapper.props(), 'closeModal');
    const cancelButton = wrapper.find('#cancel');
    cancelButton.simulate('click');
    expect(handleCancel).toHaveBeenCalledTimes(1);
    expect(closeModal).toHaveBeenCalledTimes(1);
  });

  it('should push router to edit page with conditionId when\'handleOnSend\' is called', () =>{
    const wrapper = mount(<SingleReminderDetails {...props} shouldOpen />);
    const handleOnSend = jest.spyOn(wrapper.instance(), 'handleOnSend');
    const editButton = wrapper.find('#submit');
    editButton.simulate('click');
    expect(handleOnSend).toHaveBeenCalledTimes(1);
    expect(props.history.push)
      .toHaveBeenCalledWith('/settings/reminders/edit/1');
  });
});
