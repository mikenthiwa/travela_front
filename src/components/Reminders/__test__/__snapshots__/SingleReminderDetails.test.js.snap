// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleReminderDetails /> should render correctly 1`] = `
<div
  className="reminder__deatails"
>
  <div
    className="overlay invisible  "
    onClick={null}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="modal invisible template-details"
      id=""
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      style={
        Object {
          "maxWidth": "504px",
        }
      }
      tabIndex="0"
    >
      <div
        className="modal-title-bar"
      >
        <div
          className="modal-title-text"
        >
          <div
            className="modal-title modal-title-word"
          >
            Passport Expiry Reminder
          </div>
          <div
            className="lable-text modal-bar"
          >
            <div />
          </div>
        </div>
        <button
          className="modal-close"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="close"
            src="close.svg"
          />
        </button>
      </div>
      <div
        className="modal-content"
        id=""
      >
        <div
          className="reminder_details"
        >
          <div>
            <p
              className="frequncy_count"
            >
              Reminder
               
              1
            </p>
            <div
              className="reminder__condition"
            >
              <p>
                Passport Expiry
              </p>
              <span
                className="frequency"
              >
                To be sent 2 Months to Expiry
              </span>
            </div>
          </div>
          <div>
            <p
              className="frequncy_count"
            >
              Reminder
               
              2
            </p>
            <div
              className="reminder__condition"
            >
              <p>
                Passport Expiry
              </p>
              <span
                className="frequency"
              >
                To be sent 3 Weeks to Expiry
              </span>
            </div>
          </div>
        </div>
        <hr />
        <div
          className="submit-area"
        >
          <fieldset>
            <div
              className="submit-area"
            >
              <button
                className="bg-btn bg-btn--inactive"
                id="cancel"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="bg-btn bg-btn--active"
                disabled={false}
                id="submit"
                onClick={[Function]}
                type="submit"
              >
                <undefined />
              </button>
            </div>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
